src/workflows/burn.ts

```diff
@@ -7,10 +7,16 @@ import {
 import { serializeSignature, sign, signRaw } from '../utils';
 import { BurnAddress } from './constants';
 import { GetSignableBurnRequest } from './types';
-import { StarkWallet } from '../types';
+import { StarkWallet, WalletConnection } from '../types';
 
+type burnWorkflowWithSignerRequest = WalletConnection & {
+  request: GetSignableBurnRequest;
+  transfersApi: TransfersApi;
+};
+
+/** @deprecated */
 export async function burnWorkflow(
-  signer: Signer,
+  signer: L1Signer,
```

Comment:
let's keep the type of the deprecated function

Reply:
yes

Reply:
My bad, I did that for a purpose I changed after and forgot to put it back in the original state.
Done.